File: src/api/assistant.ts
import request from './request'

export interface AssistantSource {
  id: string
  title: string
  excerpt: string
}

export interface AssistantReply {
  answer: string
  sources: AssistantSource[]
  suggested_actions: string[]
}

export async function askAssistant(question: string, page: string, imageDataUrl?: string): Promise<AssistantReply> {
  const response = await request.post('/assistant/chat', { question, page, image_data_url: imageDataUrl })
  // `request` 的响应拦截器已经返回了解包后的业务对象；此处不能再次读取 `.data`。
  return response as unknown as AssistantReply
}

File: backend/app/models/assistant.py
from __future__ import annotations

from pydantic import BaseModel, Field


class AssistantSource(BaseModel):
    id: str
    title: str
    excerpt: str


class AssistantChatRequest(BaseModel):
    question: str = Field(min_length=1, max_length=4000)
    page: str = Field(default="", max_length=120)
    image_data_url: str | None = Field(default=None, max_length=14_000_000)


class AssistantChatResponse(BaseModel):
    answer: str
    sources: list[AssistantSource]
    suggested_actions: list[str]


class AssistantUploadResponse(BaseModel):
    document_id: str
    filename: str
    indexed: bool
    message: str
